sway/desktop/output.c

```diff
@@ -746,13 +746,15 @@ static void damage_child_views_iterator(struct sway_container *con,
 void output_damage_whole_container(struct sway_output *output,
 		struct sway_container *con) {
 	int shadow_sigma = con->shadow_enabled ? config->shadow_blur_sigma : 0;
+	int blur_size = con->blur_enabled ? config_get_blur_size() : 0;
```

Comment:
Seems like we're missing some damage. When ran with damage=highlight, the damage isn't extended some of the time:

![image](https://github.com/WillPower3309/swayfx/assets/35975961/72696d9b-e11b-40c1-82ed-ab52081a96e4)

Resulting in this clipping

![image](https://github.com/WillPower3309/swayfx/assets/35975961/988b8cd7-0e90-4b14-bfda-8e3c5f8a33df)

I'll try to look into what's the cause of this

Reply:
Moving the damage expansion into `damage_surface_iterator` should do the trick:
```diff
diff --git a/sway/desktop/output.c b/sway/desktop/output.c
index 3b4a5303..efb95b90 100644
--- a/sway/desktop/output.c
+++ b/sway/desktop/output.c
@@ -689,6 +689,19 @@ static void damage_surface_iterator(struct sway_output *output,
 			ceil(output->wlr_output->scale) - surface->current.scale);
 	}
 	pixman_region32_translate(&damage, box.x, box.y);
+	if (view) {
+		struct sway_container *con = view->container;
+		int shadow_sigma = con->shadow_enabled ? config->shadow_blur_sigma : 0;
+		int blur_size = con->blur_enabled ? config_get_blur_size() : 0;
+		int effect_size = MAX(shadow_sigma, blur_size);
+
+		wlr_region_expand(&damage, &damage, effect_size);
+
+		box.x -= effect_size;
+		box.y -= effect_size;
+		box.width += effect_size * 2;
+		box.height += effect_size * 2;
+	}
 	if (wlr_damage_ring_add(&output->damage_ring, &damage)) {
 		wlr_output_schedule_frame(output->wlr_output);
 	}
@@ -745,16 +758,12 @@ static void damage_child_views_iterator(struct sway_container *con,
 
 void output_damage_whole_container(struct sway_output *output,
 		struct sway_container *con) {
-	int shadow_sigma = con->shadow_enabled ? config->shadow_blur_sigma : 0;
-	int blur_size = con->blur_enabled ? config_get_blur_size() : 0;
-	int effect_size = MAX(shadow_sigma, blur_size);
-
 	// Pad the box by 1px, because the width is a double and might be a fraction
 	struct wlr_box box = {
-		.x = con->current.x - output->lx - 1 - effect_size,
-		.y = con->current.y - output->ly - 1 - effect_size,
-		.width = con->current.width + 2 + effect_size * 2,
-		.height = con->current.height + 2 + effect_size * 2,
+		.x = con->current.x - output->lx - 1,
+		.y = con->current.y - output->ly - 1,
+		.width = con->current.width + 2,
+		.height = con->current.height + 2,
 	};
 	scale_box(&box, output->wlr_output->scale);
 	if (wlr_damage_ring_add_box(&output->damage_ring, &box)) {
```